portfolio/src/main/webapp/script.js

```diff
@@ -285,18 +310,145 @@ async function deleteCommentHistory() {
 function presentPopupCommentReceipt() {
   // Obtain user input comment content and name.
   const formElements = document.getElementById('comment-form').elements;
-  const userComment = formElements[0].value;
-  const userName = formElements[1].value;
+  const userPlaceName = formElements[0].value;
+  const userComment = formElements[1].value;
+  const userName = formElements[2].value;
 
   // Construct user comment receipt.
   const receipt = `Dear ${userName},\nThank you for submitting feedback!\n` +
                   `You entered the following:\n` +
-                  `    *Message: "${userComment}"\n`;
+                  `    *Place Name: ${userPlaceName}\n` +
+                  `    *Comment: "${userComment}"\n`;
 
   // Present comment receipt in a pop-up window.
   window.alert(receipt);
 }
 
+/**
+ * Asynchronously adds a Google map and add information regarding places
+ * referenced in the comments through associated markers and information
+ * windows.
+ */
+function addMap() {
```

Comment:
It'd be neat to have functionality to recenter the map on a pin and open the pin info if you click on the corresponding comment (right now it seems to be centered on the first comment of the current page).

Reply:
This sounds cool. I'll think about how to implement this!

Reply:
This is done! The functionality also works after you re-enable the comment history section using the checkbox (when the map and markers are reloaded but the HTML is restored from earlier, I re-linked the comments to the new markers and info windows).